daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs

```diff
@@ -81,6 +84,9 @@ required msg = fromMaybeM (throwIO $ DamlHelperError msg Nothing)
 requiredE :: Exception e => T.Text -> Either e t -> IO t
 requiredE msg = fromRightM (throwIO . DamlHelperError msg . Just . T.pack . displayException)
 
+defaulting :: a -> Either ConfigError (Maybe a) -> IO a
+defaulting a e = fmap (fromMaybe a) $ either throwIO return e
```

Comment:
Prefer `requiredE` which is defined above, results in better error messages.

Reply:
There is something odd about using `requiredE` for non required config.

Reply:
It throws an error on the error part, not the config part ... just like `defaulting`.

Reply:
I was having trouble seeing what config error could occur... given that the requirement for the field is optional. But I suppose the field might exist but be of the wrong type...

Which now makes more sense of the exiting code...
```
 mbScenario :: Maybe String <-
        requiredE "Failed to parse scenario" $
        queryProjectConfig ["scenario"] projectConfig
```

ok, I will change to be the same.

Reply:
done